src/base/bittorrent/torrentinfo.cpp

```diff
@@ -305,10 +293,22 @@ int BitTorrent::TorrentInfo::fileIndex(const QString& fileName) const
     return -1;
 }
 
-void TorrentInfo::remapFiles(libtorrent::file_storage const &fileStorage)
+void TorrentInfo::stripRootFolder()
 {
-    if (!isValid()) return;
-    m_nativeInfo->remap_files(fileStorage);
+    if (filesCount() <= 1) return;
+
+    libtorrent::file_storage files = m_nativeInfo->files();
+
+    // Solution for case of renamed root folder
+    std::string testName = filePath(0).split('/').value(0).toStdString();
+    if (files.name() != testName) {
+        files.set_name(testName);
+        for (int i = 0; i < files.num_files(); ++i)
+            files.rename_file(i, files.file_path(i));
```

Comment:
As far as I remember when you rename root folder in AddNewTorrentDialog the files become independent from storage (since it renamed via rename_file and storage still has its original name). We bind it together here.